### DISCUSSION TITLE
Hidden files within Telescope

### DISCUSSION THREAD

**Discussion by ShadowMaxx122** [LazyVim/LazyVim#2009] [Q&A] [answered]
Hello!

I am a new neovim user coming from vanilla vim. I have tried for the last 2 hours how to figure out how to show hidden files in telescope and I can’t make anything work. Is there something different that has to be done for LazyVim?

**Comment by dpetka2001** [comment]
When you have Telescope open you can press `<C-/` to show you what key actions are available for the current picker. Lazyvim has provided a shortcut `<M-h>` for searching with hidden files enabled.

**Reply by ShadowMaxx122** [reply]
Oh I didn’t realize that. Is there a special way to make it search with hidden files by default?

**Reply by dpetka2001** [reply, marked as answer]
You would have to create either a new keymap invoking telescope with the options you want or maybe try to change the options for the picker you're interested in. Keep in mind that's not the default telescope behavior, so Lazyvim doesn't change the default option and instead provides a keymap for that.

**Reply by ShadowMaxx122** [reply]
Ah, that’s where I was going wrong! I was trying to configure telescope. Ok, that’s a lot of help! Thank you!

**Reply by dpetka2001** [reply]
You could do it in Telescope too. But your custom keymap would be easier.

**Comment by dimas-cyriaco** [comment]
Exemple of setting live grep to include hidden files (this was very trick to find out... 😅 ):

```lua
vim.keymap.set(
  "n",
  "<leader>sg",
  ":lua require('telescope.builtin').live_grep({ additional_args = function() return { '--hidden' } end })<cr>",
  { silent = true, desc = "Find in Files" }
)
```

**Reply by dpetka2001** [reply]
Why do you use a function here, if all you do is return the string in a table in `additional_args`? You could directly pass the string in a table.

**Reply by dimas-cyriaco** [reply]
That's one of the reason it was very tricky to get this solution. 😂
For some reason, passing the argument directly does not work (at least for me).

**Reply by dimas-cyriaco** [reply]
I suppose the function is passed as a parameter and not executed right away, and the `live_grep` function calls `addictional_args()` as a function and use it's return.

**Reply by dpetka2001** [reply]
```lua
vim.keymap.set(
  "n",
  "<leader>sg",
  "<cmd>lua require('telescope.builtin').live_grep({additional_args = {'--hidden'}})<cr>"
)
```
this worked just fine for me.

**Reply by dimas-cyriaco** [reply]
Now that I copied and pasted your code I realized I was missing the `{ }` around `--hidden`... 🤦‍♂️ 
Thank you! 😅

**Reply by dpetka2001** [reply]
Keep in mind you will have to disable the corresponding Telescope keymap in `keys` section of Telescope spec. It will only work the first time and not afterwards, because Telescope will load after the first time and will overwrite your keymap. Assuming you keep the keymap `<leader>sg` and not change to something else.
